### DISCUSSION TITLE
Create New Node Search filter should have an option to display full tree on the selected node

### DISCUSSION THREAD

**Discussion by wyattbiker** [godotengine/godot-proposals#5258] [Editor]
When creating a new node, I search to filter in nodes containing a string in the name .E.g text. Many times I want to find out what is around a particular node from the list shown. So I highlighted and then I have to clear the filter and remember the tree to navigate through to find it and see what is around it.

Would be nice to have a button, that just simply shows the complete tree while the cursor highlight remains on that node.

E.g.

Go from this:

![image](https://user-images.githubusercontent.com/930478/186740452-b93b430b-37fa-41d3-9fe8-191c19c07c5b.png)

to this:

![image](https://user-images.githubusercontent.com/930478/186741085-20ae691e-dfd0-41fa-a9ff-4f57a8a9dd88.png)

It can be done either one of two ways. 

1. Have a checkbox to display the full tree while keeping the highlighted node in place. Unchecking the checkbox redisplays the filter with node highlighted.

3. A simple way would be When the search filter is Xed out, show the full tree but keep it on highlighted node (simpler but loss search filter) but I prefer option 1

**Comment by bcmpinc** [comment]
I don't know how 1 would work in practice. Would love to see option 2 implemented though. I'm always annoyed that when I clear the search to show the full tree it deselects the item I had selected. The fact that double clicking items to unfold them closes the window only adds to the frustration. Sometimes I opt for navigating by keyboard, but as you cannot select greyed out items, you cannot unfold those using the keyboard.

**Comment by actforjason** [comment]
Maybe put "show tree" as a button on the right side of the input box instead of below it to avoid taking up too much space.
